src/api-requests/get-all-meta-info.ts

```diff
@@ -1,11 +1,77 @@
 import { UnderlyingClient } from '../types/client-factory';
 import { ApolloQueryResult, gql } from '@apollo/client';
 import { MetaInfoResponse } from '../interfaces/meta-info';
+import { isStrapiClient } from '../utils/client-factory';
+import { TagType } from '../types/tag-type';
+import { insertTypename, extractFieldsFromNode, areTheSame } from './utils';
+import { CS_CLIENT, STRAPI_CLIENT } from '../config/api-client';
+
+type metaInfoType =
+    | 'l1Products'
+    | 'l2Products'
+    | 'programmingLanguages'
+    | 'technologies'
+    | 'expertiseLevels'
+    | 'contentTypes';
+
+/**
+ * Ensure CS response is compatible with original Strapi format
+ */
+const formatResponse = (
+    isStrapiClient: boolean,
+    metaInfoType: metaInfoType,
+    data: any,
+    __typename: TagType
+) => {
+    let formattedData = data[metaInfoType];
+
+    if (isStrapiClient) {
+        return formattedData;
+    }
+
+    formattedData = formattedData.items;
+
+    // explicitly define default when falsy
+    formattedData = formattedData.map((d: any) => ({
+        name: d.name ? d.name : null,
+        description: d.description ? d.description : null,
+        slug: d.slug ? d.slug : null,
+        secondary_cta: d.secondary_cta ? d.secondary_cta : '',
+        primary_cta: d.primary_cta ? d.primary_cta : '',
+        documentation_link: d.documentation_link ? d.documentation_link : null,
+    }));
+
+    if (metaInfoType === 'l2Products') {
+        formattedData = formattedData.map((d: any) => ({
+            ...d,
+            l1_product: {
+                l_1_product: extractFieldsFromNode(d.l1_product, ['name']),
+            },
+        }));
+    }
+
+    return insertTypename(formattedData, __typename);
+};
 
 export const getAllL1ProductsMetaInfo = async (
-    client: UnderlyingClient<'ApolloREST'>
+    client: UnderlyingClient<'ApolloREST'> | UnderlyingClient<'ApolloGraphQL'>
```

Comment:
While I cannot conclude even after some research, I think the high-level differences come down to network latency and runtime overheads.
- General-purpose HTTP libraries are more lightweight, so they have lower runtime overheads than ApolloClients'
- ApolloClient is tailored for GraphQL requests, so it should reduce network requests with the help of caching and other optimizations

I am not sure of the net effects between the two, but my guess is that network latency will be a bigger bottleneck than execution overheads.

@mcolella14 Could you opine on this and share your rationale behind using `axios`?

Reply:
Yeah I was against using Apollo Client because we really don't need it and it adds overhead. I believe we used it to mimic GraphQL queries to Strapi initially, but since we have an actual GraphQL endpoint that is easy to use, I don't think we really need it.

In-memory caching is an advantage though, and there may be some performance advantages, but I'm not really sure.

I'm fine with keeping Apollo if we want, but it's use case really seems to be geared to using in React components, and there are other alternatives to caching like https://www.npmjs.com/package/axios-cache-interceptor.